src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/oops/ConstMethod.java

```diff
@@ -429,18 +429,74 @@ private boolean hasMethodAnnotations() {
     return (getFlags() & HAS_METHOD_ANNOTATIONS) != 0;
   }
 
+  public U1Array getMethodAnnotations() {
+    if (hasMethodAnnotations()) {
+      Address addr = getAddress().getAddressAt((getSize() - 1) * VM.getVM().getAddressSize());
+      return VMObjectFactory.newObject(U1Array.class, addr);
+    } else {
+      return null;
+    }
+  }
+
   private boolean hasParameterAnnotations() {
     return (getFlags() & HAS_PARAMETER_ANNOTATIONS) != 0;
   }
 
-  private boolean hasDefaultAnnotations() {
-    return (getFlags() & HAS_DEFAULT_ANNOTATIONS) != 0;
+  public U1Array getParameterAnnotations() {
+    if (hasParameterAnnotations()) {
+      int offset = 1;
+      if (hasMethodAnnotations()) {
+        offset += 1;
+      }
+      Address addr = getAddress().getAddressAt((getSize() - offset) * VM.getVM().getAddressSize());
```

Comment:
A comment on the address computation would be useful here and in the changes below.

Reply:
Added a comment about the layout of the annotation pointers.